File: JavaScript/Algorithms/Recursion/recursion.js
// 1. Identify the base case
// 2. Identify the recursive case
// 3. Return when needed

let counter = 0;
function inception() {
  console.log('counter: ', counter)
  if (counter > 3) {
    return console.log('Done')
  }
  counter++;
  return inception();
}

inception();

File: Python/Algorithms/Recursion/recursion.py
# 1. Identify the base case
# 2. Identify the recursive case
# 3. Return when needed

counter = 0
def inception():
    global counter
    print('counter: ')
    print(counter)
    if counter > 3:
        print('Done')
        return
    counter += 1
    return inception()


inception()
